src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -537,9 +551,17 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
                 }
 
                 // Buffer the response content if we've been asked to and we have a Content to buffer.
-                if (response.Content != null)
+                if (buffered && response.Content != null)
```

Comment:
Should it also check: `response.Content is not EmptyContent` ?

After updating an app from .NET Core 3.1 to .NET 5.0, I am hitting ObjectDisposedException:

```sh
  ---- System.ObjectDisposedException : Cannot access a disposed object.
  Object name: 'System.Net.Http.EmptyContent'.
    Stack Trace:
       at System.Net.Http.HttpContent.CheckDisposed()
     at System.Net.Http.HttpContent.LoadIntoBufferAsync(Int64 maxBufferSize, CancellationToken cancellationToken)
     at System.Net.Http.HttpClient.SendAsyncCore(HttpRequestMessage request, HttpCompletionOption completionOption, Boolean async, Boolean emitTelemetryStartStop, CancellationToken cancellationToken)
```

Reply:
Nvm, it was a bug in test, which 3.1 was tolerating and 5.0 makes sure the objects are disposed properly: https://github.com/actions/runner/pull/799/files#diff-f7999ffa872c1d85e7368f4c351978e3d2a4ad3e3b9a27547234ede8e26bdaf5.